crucible-mir/src/Mir/Trans.hs

```diff
@@ -149,33 +149,64 @@ transConstVal _ty (Some (UsizeRepr)) (M.ConstInt i) =
        return $ MirExp UsizeRepr (S.app $ usizeLit n)
 transConstVal _ty (Some (IsizeRepr)) (ConstInt i) =
       return $ MirExp IsizeRepr (S.app $ isizeLit (fromIntegerLit i))
-transConstVal (M.TyRef (M.TySlice ty) _) (Some (MirSliceRepr tpr)) (M.ConstSlice cs) = do
-    cs' <- Trav.for cs $ \c -> do
-        MirExp tpr' c' <- transConstVal ty (Some tpr) c
-        Refl <- testEqualityOrFail tpr tpr' $
-            "transConstVal (ConstSlice): returned wrong type: expected " ++
-            show tpr ++ ", got " ++ show tpr'
-        pure c'
-    vec <- mirVector_fromVector tpr $ R.App $ E.VectorLit tpr $ V.fromList cs'
-    vecRef <- constMirRef (MirVectorRepr tpr) vec
-    ref <- subindexRef tpr vecRef (R.App $ usizeLit 0)
-    let len = R.App $ usizeLit $ fromIntegral $ length cs
-    let struct = S.mkStruct
-            (mirSliceCtxRepr tpr)
-            (Ctx.Empty Ctx.:> ref Ctx.:> len)
-    return $ MirExp (MirSliceRepr tpr) struct
-transConstVal _ty (Some (MirSliceRepr u8Repr@(C.BVRepr w))) (M.ConstStr bs)
+
+--
+-- Before the slice code got split into separate body and reference steps,
+-- it did the following after the code that's equivalent to the ConstArray
+-- case (which is where constant slice bodies get handled now):
+--
+--vecRef <- constMirRef vec_tpr vec
+--ref <- subindexRef tpr vecRef (R.App $ usizeLit 0)
+--let len = R.App $ usizeLit $ fromIntegral $ length cs
+--let struct = S.mkStruct (mirSliceCtxRepr tpr) (Ctx.Empty Ctx.:> ref Ctx.:> len)
+--return $ MirExp (MirSliceRepr tpr) struct
+--
+-- Therefore, after looking up the defid, the reference step needs to:
+--    * extract the type from the global variable it finds
+--      (note that it'll be a MirVectorRepr that it needs to unwrap)
+--    * construct a reference to the global variable
+--      (with globalMirRef rather than constMirRef, that's the point of
+--      all this)
+--    * apply subindexRef as above
+--    * cons up the length
+--    * call mkStruct
+--    * cons up the final MirExp
+--
+-- staticPlaces does the first four of these actions; addrOfPlace
+-- does the last two. Note that addrOfPlace uses mkSlice instead
+-- of mkStruct as above, but as far as I can tell it's equivalent.
```

Comment:
Yes, slices are encoded as structs with very particular field types, so it's perfectly fine to use `mkSlice` in place of `mkStruct`.

Reply:
Yes, the question was whether it's exactly the same or if there was something special/different about the mkStruct call such that it wasn't using mkSlice for a reason. Which there apparently wasn't. But it wasn't obvious without expanding/looking up a bunch of things.